lunatrace/bsl/backend/src/graphql-yoga/resolvers/create-pull-request-for-vulnerability.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright by LunaSec (owned by Refinery Labs, Inc)
+ *
+ * Licensed under the Business Source License v1.1
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * https://github.com/lunasec-io/lunasec/blob/master/licenses/BSL-LunaTrace.txt
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { GraphQLYogaError } from '@graphql-yoga/node';
+import {
+  PullRequestOctokit,
+  replacePackageAndFileGitHubPullRequest,
+} from '@lunatrace/npm-package-cli/src/package/github-pr';
+
+import { getInstallationAccessToken } from '../../github/auth';
+import { hasura } from '../../hasura-api';
+import { log } from '../../utils/log';
+import { MutationResolvers } from '../generated-resolver-types';
+import { checkProjectIsAuthorizedOrThrow, isAuthenticated } from '../helpers/auth-helpers';
+
+type CreatePullRequestForVulnerabilityType = NonNullable<MutationResolvers['createPullRequestForVulnerability']>;
+
+function splitGitHubRepoPath(gitHubRepo: string) {
+  const split = gitHubRepo.split('/');
+  const owner = split[split.length - 2];
+  const repo = split[split.length - 1].replace('.git', '');
+  return { owner, repo };
+}
+
+/**
+ * Installs the repos the user selected in the GUIG
+ */
+export const createPullRequestForVulnerabilityResolver: CreatePullRequestForVulnerabilityType = async (
+  parent,
+  args,
+  ctx,
+  _info
+) => {
+  try {
+    if (!isAuthenticated(ctx)) {
+      log.warn('No parsed JWT claims with a user ID on route that required authorization, throwing a graphql error');
+      throw new GraphQLYogaError('Unauthorized');
+    }
+
+    const vulnerabilityId = args.vulnerability_id;
+    if (!vulnerabilityId) {
+      throw new GraphQLYogaError('No vulnerability id provided');
+    }
+
+    const projectId = args.project_id;
+    if (!projectId) {
+      throw new GraphQLYogaError('No project id provided');
+    }
+
+    const oldPackageSlug = args.old_package_slug;
+    if (!oldPackageSlug) {
+      throw new GraphQLYogaError('No old package slug provided');
+    }
+
+    const newPackageSlug = args.new_package_slug;
+    if (!newPackageSlug) {
+      throw new GraphQLYogaError('No new package slug provided');
```

Comment:
We could probably DRY up this to just be a param checking function that takes an array of property names as strings. I bet we could throw that in utils and call it a day. 


Just a thought, not like an urgent change obv